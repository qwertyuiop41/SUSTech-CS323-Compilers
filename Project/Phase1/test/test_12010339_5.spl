struct friends
int main()
{
int a=4-2;
	
}
int t(){
	if(1+2*7+3==2){
    return 0;
  }
}