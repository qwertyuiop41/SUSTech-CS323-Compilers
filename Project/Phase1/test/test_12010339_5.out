Program (1)
  ExtDefList (1)
    ExtDef (1)
      Specifier (1)
        TYPE: int
      ExtDecList (1)
        VarDec (1)
          ID: global
      SEMI
    ExtDefList (2)
      ExtDef (2)
        Specifier (2)
          StructSpecifier (2)
            STRUCT
            ID: my_struct
            LC
            DefList (4)
              Def (4)
                Specifier (4)
                  TYPE: int
                DecList (4)
                  Dec (4)
                    VarDec (4)
                      ID: code
                SEMI
              DefList (5)
                Def (5)
                  Specifier (5)
                    TYPE: char
                  DecList (5)
                    Dec (5)
                      VarDec (5)
                        ID: data
                  SEMI
            RC
        SEMI
      ExtDefList (7)
        ExtDef (7)
          Specifier (7)
            TYPE: int
          FunDec (7)
            ID: test
            LP
            RP
          CompSt (8)
            LC
            DefList (9)
              Def (9)
                Specifier (9)
                  StructSpecifier (9)
                    STRUCT
                    ID: my_struct
                DecList (9)
                  Dec (9)
                    VarDec (9)
                      ID: obj
                SEMI
            StmtList (10)
              Stmt (10)
                Exp (10)
                  Exp (10)
                    Exp (10)
                      ID: obj
                    DOT
                    ID: code
                  ASSIGN
                  Exp (10)
                    ID: global
                SEMI
              StmtList (11)
                Stmt (11)
                  Exp (11)
                    Exp (11)
                      ID: global
                    ASSIGN
                    Exp (11)
                      Exp (11)
                        ID: global
                      PLUS
                      Exp (11)
                        Exp (11)
                          Exp (11)
                            Exp (11)
                              INT: 1
                            MUL
                            Exp (11)
                              INT: 4
                          MUL
                          Exp (11)
                            INT: 3
                        DIV
                        Exp (11)
                          INT: 2
                  SEMI
            RC
        ExtDefList (13)
          ExtDef (13)
            Specifier (13)
              TYPE: int
            FunDec (13)
              ID: t
              LP
              RP
            CompSt (13)
              LC
              StmtList (14)
                Stmt (14)
                  IF
                  LP
                  Exp (14)
                    Exp (14)
                      ID: a
                    EQ
                    Exp (14)
                      Exp (14)
                        Exp (14)
                          INT: 8
                        DIV
                        Exp (14)
                          INT: 2
                      MUL
                      Exp (14)
                        INT: 3
                  RP
                  Stmt (14)
                    CompSt (14)
                      LC
                      StmtList (15)
                        Stmt (15)
                          RETURN
                          Exp (15)
                            INT: 0
                          SEMI
                      RC
              RC
